### PR TITLE
Bug Fix, Add Rubocop, Add TravisCI, Adds Atrium::Connect, Increases spec coverage

### LINKED ISSUE #14: Support connect widget
Add in some helper methods for the connect widget

### PR DESCRIPTION
**Pull request by sirwolfgang**
- Adds and Applies rubocop, bring the repo more in line with community standards. 
- Adds Travis-ci to the project. All that is missing is for a collaborator to enable it on https://travis-ci.org Completes #27
- Adds test coverage for the client
- Connect widget support, Closes #14
- Fixes bug with client being unable to build empty body requests
- Adds VCR in order to start supporting end to end spec coverage
- Adds support for the [read institution endpoint](https://atrium.mx.com/documentation#read-institution)

Note: The tests as is, are heavily mocked leaving the client to be untested. This allowed a bug with empty request body building to go on untested. This PR fixes that issue, adds mocked coverage to client, and adds end to end coverage of a few of the user specs which are where I located this bug. The rest of the specs should be reflowed to leverage VCR.

This combines the #28, #29, #30 pull requests.

### CODE REVIEW

**Comment by sirwolfgang** [comment]
@film42 Can I get a review on this?

**Comment by sirwolfgang** [comment]
The requested change has been made.

File: lib/atrium/institution.rb [hunk 1]
```diff
@@ -8,7 +8,10 @@ class Institution
     attribute :name
     attribute :small_logo_url
     attribute :url
+    attribute :medium_logo_url
+    attribute :small_logo_url
```
Comment (film42) [inline lib/atrium/institution.rb]: Looks like this wasn't rebase correctly. Both small and medium uris are duplicated here.

File: spec/lib/atrium/institution_spec.rb [hunk 2]
```diff
@@ -86,4 +87,36 @@
...
+    it "should return an array of institutions" do
+      response = ::Atrium::Institution.list
+
+      expect(response).to be_kind_of(::Array)
+      expect(response.length).to eq(4)
+      expect(response.first).to be_kind_of(::Atrium::Institution)
+
+      expect(response.first.code).to eq(institution_attributes[:code])
+      expect(response.first.name).to eq(institution_attributes[:name])
+      expect(response.first.url).to eq(institution_attributes[:url])
```
Comment (film42) [inline spec/lib/atrium/institution_spec.rb]: I think the small and medium logo uri tests in master were deleted.

File: spec/support/vcr.rb [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+RSpec.configure do |config|
+  # Add VCR to all tests
+  config.around(:each) do |example|
+    options = example.metadata[:vcr] || {}
+
+    if options[:record] == :skip
+      VCR.turned_off(&example)
+    else
+      name = example
```
Comment (film42) [inline spec/support/vcr.rb]: I'm not normally into fancy test libs, but VCR looks cool. What's up with this long name formatting?

Comment (sirwolfgang) [inline spec/support/vcr.rb]: By default, VCR requires manual management of the cassettes. This becomes a huge stateful pain. This bit of magic takes all the `describe`, `context`, and `it` strings, binds them together into a file path/tree. Allowing the automatic creation and management of cassettes in any and all situations. The vast majority of this is to make the folder and file names valid.

**Comment by sirwolfgang** [comment]
The requested change has been made.

File: spec/support/vcr.rb (hunk 3 above)
Comment (film42) [inline spec/support/vcr.rb]: Is there something explicit we can add to each spec file that uses VCR? Looks like there are only 4 cassettes. As is, this is certainly not very maintainable.

Comment (sirwolfgang) [inline spec/support/vcr.rb]: Not really, this VCR hooks into webmock and intercepts all network requests. This is so you don't have to mock out the client, and can actually test your code. The rest of the specs need to be updated to use VCR. As it is, the completely mocked out, and untested, client contains bugs due to this incomplete usage.

Most of the specs test the mocks and not the code. Which is how, and why, I discovered this bug: https://github.com/mxenabled/atrium-ruby/pull/31/commits/90211ae051273525fc0da10d1dcf1afa231a3193

Comment (sirwolfgang) [inline spec/support/vcr.rb]: Feel free to cherry pick the bug fixes, or add to the PR in whatever way you see fit.

**Comment by film42** [comment]
@sirwolfgang I made a few changes to casette loading. @ztoolson mind giving a final review?

**Comment by ztoolson** [comment]
:shipit: Changes look good. Thanks @film42 and @sirwolfgang .

**Comment by film42** [comment]
Thanks @sirwolfgang ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️

**Comment by film42** [comment]
Released as `0.5.0`. Doesn't really need a minor version bump, but seemed safer than a patch version bump.

**Pull request merged**